### PR TITLE
add bmpstr support

### PR DESCRIPTION
**Pull request by fleg**
tested on node v0.10.40 and v5.1.0

### CODE REVIEW

**Comment by fleg** [comment]
I'm also saw bmpstr implementation in #47, it uses ucs2 (utf16le), but bmpstr should use utf16be (like [here](https://github.com/lapo-luchini/asn1js/blob/015c5057789f37b72a68182012a30e2d334fca02/asn1.js#L93-L101)), am I right?

File: lib/asn1/decoders/der.js [hunk 1]
```diff
@@ -133,6 +133,13 @@ DERNode.prototype._decodeStr = function decodeStr(buffer, tag) {
     return { unused: unused, data: buffer.raw() };
   } else if (tag === 'ia5str' || tag === 'utf8str') {
     return buffer.raw().toString();
+  } else if(tag === 'bmpstr') {
```
Comment (indutny) [inline lib/asn1/decoders/der.js]: Should it verify that the length is divisible by 2?

File: lib/asn1/encoders/der.js [hunk 2]
```diff
@@ -67,6 +67,13 @@ DERNode.prototype._encodeStr = function encodeStr(str, tag) {
     return this._createEncoderBuffer([ str.unused | 0, str.data ]);
   else if (tag === 'ia5str' || tag === 'utf8str')
     return this._createEncoderBuffer(str);
+  else if (tag === 'bmpstr') {
+    var buf = new Buffer(str.length * 2);
+    for (var i = 0; i < str.length; ++i) {
```
Comment (indutny) [inline lib/asn1/encoders/der.js]: `i++`, please.

File: lib/asn1/decoders/der.js [hunk 3]
```diff
@@ -133,6 +133,13 @@ DERNode.prototype._decodeStr = function decodeStr(buffer, tag) {
...
+    var raw = buffer.raw();
+    var str = '';
+    for (var i = 0; i < raw.length / 2; ++i) {
```
Comment (indutny) [inline lib/asn1/decoders/der.js]: `i++`

**Comment by indutny** [comment]
@fleg do you have any use case for this? Perhaps some DER encoded data that could be included in a test?

**Comment by indutny** [comment]
Few nits, otherwise LGTM. Thank  you!

**Comment by fleg** [comment]
fixed

**Comment by fleg** [comment]
I have this [cert](https://gist.github.com/fleg/c878f33d09397c94a05b)
Almost all strings in issuer filed have `bmpstr` encoding, it also has unsupported `numstr` and `printstr` strings.

**Comment by indutny** [comment]
Thank you! Landed and released in 4.1.0.

**Pull request closed without merging**